code/datums/dna.dm

```diff
@@ -1,3 +1,41 @@
+///Setter macro used to modify dna blocks. For convenience, use SET_UI_BLOCK for identity and SET_UF_BLOCK for features.
+#define SET_DNA_BLOCK(string, blocknum, input, len_by_block) string = copytext(string, 1, len_by_block[blocknum]) + input + (blocknum < len_by_block.len ? copytext(string, len_by_block[blocknum+1]) : 0)
+
+#define SET_UI_BLOCK(string, blocknum, input) SET_DNA_BLOCK(string, blocknum, input, GLOB.total_ui_len_by_block)
+#define SET_UF_BLOCK(string, blocknum, input) SET_DNA_BLOCK(string, blocknum, input, GLOB.total_uf_len_by_block)
+
+/**
+ * List of unique indentity blocks whose length is different than the default dna block size.
+ * IF YOU ADD ANY UNIQUE IDENTITY BLOCK WHOSE SIZE IS DIFFERENT THAN DNA_BLOCK_SIZE, INCLUDE IT HERE.
+ */
+GLOBAL_LIST_INIT(identity_block_lengths, list(
+		"[DNA_HAIR_COLOR_BLOCK]" = DNA_BLOCK_SIZE_COLOR,
+		"[DNA_FACIAL_HAIR_COLOR_BLOCK]" = DNA_BLOCK_SIZE_COLOR,
+		"[DNA_EYE_COLOR_BLOCK]" = DNA_BLOCK_SIZE_COLOR,
+	))
+///Ditto but for features. IF YOU ADD ANY UNIQUE FEATURE BLOCK OF SIZE DIFFERENT THAN DNA_BLOCK_SIZE, INCLUDE IT HERE.
+GLOBAL_LIST_INIT(features_block_lengths, list(
+		"[DNA_MUTANT_COLOR_BLOCK]" = DNA_BLOCK_SIZE_COLOR,
+		"[DNA_ETHEREAL_COLOR_BLOCK]" = DNA_BLOCK_SIZE_COLOR,
+	))
+///A list containing the total length of the unique identifier dna string up to a given block (excluding the block itself).
+GLOBAL_LIST(total_ui_len_by_block)
+///Ditto but for unique features.
+GLOBAL_LIST_INIT(total_uf_len_by_block, populate_total_dna_len_by_block())
+
+///Populates both ui and uf len by block lists.
+/proc/populate_total_dna_len_by_block()
+	GLOB.total_ui_len_by_block = list()
+	var/total_block_len = 1
+	for(var/blocknumber in 1 to DNA_UNI_IDENTITY_BLOCKS)
+		GLOB.total_ui_len_by_block += total_block_len
+		total_block_len += GET_UI_BLOCK_LEN(blocknumber)
+
+	. = list()
```

Comment:
Why .? and why are we initing two global lists in one proc?

Reply:
I assume you know how GLOBAL_LIST_INIT works, the first parameter is the name of the new global list and the second is its value, in this case the return value of a proc.
I have split it in two procs anyway, so it shouldn't be a bother anymore.

Reply:
I do yes, I just don't like using . like this here